### PR TITLE
#14114: Support for personal access token

### LINKED ISSUE #14114: Enable User tokens to access APIs
**Feature**
Currently, the only way anyone can access it is through bot tokens. Most of them use ingestion-bot tokens.
However, bot tokens are limited because they can't update existing description tags, etc..
We have a user token option in the backend, and we should enable this in the UI so that one can use their user token
to access APIs.

cc @chirag-madlani @mohityadav766

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes:

Fixes #14114 

I worked on adding a support for generation of Personal Access Token

https://github.com/open-metadata/OpenMetadata/assets/58542468/e1b24ded-ef10-4177-be3f-bfbdde6e7970

#
### Type of change:

- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AccessTokenCard/AccessTokenCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    createUserAccessTokenWithPut: jest.fn().mockImplementation(() =>
+      Promise.resolve({
+        JWTTokenExpiry: 'tesst',
+        tokenName: 'test',
+      })
+    ),
+  }));
+  jest.mock('../../rest/userAPI', () => ({
+    getUserAccessToken: jest.fn().mockImplementation(() => Promise.resolve({})),
+  }));
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/AccessTokenCard/AccessTokenCard.test.tsx:38]: mock should be outside of describe block

File: openmetadata-ui/src/main/resources/ui/src/components/Users/mocks/User.mocks.ts [hunk 2]
```diff
@@ -1030,3 +1030,13 @@ export const mockEntityData: {
   total: 6,
   currPage: 1,
 };
+export const mockAccessData = {
+  expiryDate: 1701714886101,
+  jwtToken:
+    // eslint-disable-next-line max-len
+    'eyJraWQiOiJHYjM4OWEtOWY3Ni1nZGpzLWE5MmotMDI0MmJrOTQzNTYiLCJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJvcGVuLW1ldGFkYXRhLm9yZyIsInN1YiI6ImFkbWluIiwiZW1haWwiOiJhZG1pbkBvcGVubWV0YWRhdGEub3JnIiwiaXNCb3QiOmZhbHNlLCJ0b2tlblR5cGUiOiJQRVJTT05BTF9BQ0NFU1MiLCJpYXQiOjE3MDE3MTEyODYsImV4cCI6MTcwMTcxNDg4Nn0.YcA8yN6n35uwxC2bwakSN-h_uyXEtAM_Y7wbku4G4bqn0pQzbFCUxz_uZ2JGF4sQsbvi2JizhcBuu7t29d7ERT2R6kJ63TmAgxviz0e2MicVS0Fe2SW0sPLl4QnhRya2kuDYihp1xMdgAzF7csTyFS7E1tpBUJqoViVlRHUvpXCpeTn1kX023NxW2iBRt-2wGsYkRtbDrWjVsZRsIujL0Wo3pDP_JEgxBE6ffo2zzl2OfJAUh0h9AQqktkf7N5g4WkX_0mbqhX1xTdqtmuUcJaGwM1Ricz-lnNuQIzr6ABQwH8VvAOgmmL-pssoLdzelh19GZO0O0LY4ik2yFOionQ',
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/mocks/User.mocks.ts:1037]: We can use dummy token here instead some real token value

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts [hunk 3]
```diff
@@ -233,3 +234,40 @@ export const createUserWithPut = async (userDetails: CreateUser) => {
...
+
+export const getUserAccessToken = async () => {
+  const response = await APIClient.get<{
+    data: PersonalAccessToken[];
+  }>('/users/security/token');
+
+  return response.data.data;
+};
+
+export const createUserAccessTokenWithPut = async ({
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:246]:
```suggestion
export const updateUserAccessToken = async ({
```

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts [hunk 4]
```diff
@@ -233,3 +234,40 @@ export const createUserWithPut = async (userDetails: CreateUser) => {
...
+  return response.data;
+};
+
+export const revokeAccessTokenWithPut = async (params: string) => {
+  const response = await APIClient.put<PersonalAccessToken>(
+    '/users/security/token/revoke?' + params
+  );
+
+  return response.data;
+};
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:273]: I don't think we need this

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AccessTokenCard/AccessTokenCard.test.tsx (hunk 1 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/AccessTokenCard/AccessTokenCard.test.tsx:38]: Done

File: openmetadata-ui/src/main/resources/ui/src/components/Users/mocks/User.mocks.ts (hunk 2 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/Users/mocks/User.mocks.ts:1037]: Sure

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts (hunk 3 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:246]: Updated

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts (hunk 4 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts]: We would need this as BE expects params in query parameters

**Review by chirag-madlani — APPROVED** [review]

**Review by 07Himank — APPROVED** [review]

**Pull request merged**